### PR TITLE
Binary Lifting for finding LCA

### PR DESCRIPTION
**Pull request by h-sinha**
Subtask 3 of #151

### CODE REVIEW

File: Data Structures/Trees/LCA/BinaryLifting.cpp [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+	if( v1 == v2 )return v1;
+
+	for (int i = 21; i >= 0; --i)
+	{
+		// Jump higher only if the 2^i th ancestor of both the nodes are not the same. This is done because 
+		// the nodes in the path from the LCA to the root are all common ancestors of the 2 nodes and if 
+		// we need to reach the LCA we should avoid jumping to a node at a higher level than the LCA. 
+		// This condition helps us avoid it. Ultimately we reach a node which is the 1st child of 
+		// LCA and then it's easy to get the LCA as it would be the 2^0 th ancestor of the node. 
+		if(parent[v1][i] != parent[v2][i])
```
Comment (sahilbansal17) [inline Data Structures/Trees/LCA/BinaryLifting.cpp]: You can also mention that when we reach away from the root of the tree, i.e. `parent[v1][i] = -1`, jump doesn't take place and we get the correct result.

Comment (h-sinha) [reply Data Structures/Trees/LCA/BinaryLifting.cpp]: I have added the condition @sahilbansal17

Comment (sahilbansal17) [reply Data Structures/Trees/LCA/BinaryLifting.cpp]: Good work. 👍

Comment (sahilbansal17) [reply Data Structures/Trees/LCA/BinaryLifting.cpp]: By the way, the condition would be handled **implicitly**, what do you say? 
As both *v<sub>1</sub>* and *v<sub>2</sub>* will be pointing to **-1** and hence they will be equal.

Comment (h-sinha) [reply Data Structures/Trees/LCA/BinaryLifting.cpp]: That's why I didn't add that condition.But this would give a clearer explanation I guess

Comment (sahilbansal17) [reply Data Structures/Trees/LCA/BinaryLifting.cpp]: Yes, its better now.

**Pull request merged**